packages/xo-server-perf-alert/src/index.js

```diff
@@ -1,1077 +1,144 @@
-  })
+  #strategy
 
-const raiseOrLowerAlarm = (alarmId, shouldRaise, raiseCallback, lowerCallback) => {
-  const current = currentAlarms[alarmId]
-  if (shouldRaise) {
-    if (!current) {
-      currentAlarms[alarmId] = true
-      raiseCallback(alarmId)
-    }
-  } else if (current) {
-    try {
-      lowerCallback(alarmId)
-    } finally {
-      delete currentAlarms[alarmId]
-    }
+  constructor(xo) {
+    this._xo = xo
   }
-}
 
-async function getServerTimestamp(xapi, host) {
-  const serverLocalTime = await xapi.call('host.get_servertime', host.$ref)
-  return Math.floor(utcParse('%Y%m%dT%H:%M:%SZ')(serverLocalTime).getTime() / 1000)
-}
+  /**
+   *
+   * @param {AlarmChanges} param0
+   * @returns
+   */
+  async sendAlarmChange({ newAlarms, closedAlarms, activeAlarms }) {
+    if (newAlarms.size === 0 && closedAlarms.size === 0) {
+      // don't send anything is the alarms didn't change
+      return
+    }
+    const byRules = {}
+    newAlarms.forEach(alarm => {
+      byRules[alarm.rule.id] = byRules[alarm.rule.id] ?? { alarms: [] }
+      byRules[alarm.rule.id].alarms.push({
+        ...alarm,
+        url: this._generateUrl(alarm.rule.objectType, alarm.target),
+        notificationType: 'new',
+      })
+    })
 
-const isSrWritable = sr => sr !== undefined && sr.content_type !== 'iso' && sr.size > 0
+    activeAlarms.forEach(alarm => {
+      byRules[alarm.rule.id] = byRules[alarm.rule.id] ?? { alarms: [] }
+      byRules[alarm.rule.id].alarms.push({
+        ...alarm,
+        url: this._generateUrl(alarm.rule.objectType, alarm.target),
+        notificationType: 'active',
+      })
+    })
 
-class PerfAlertXoPlugin {
-  constructor(xo) {
-    this._xo = xo
-    this._job = createSchedule('* * * * *').createJob(async () => {
-      try {
-        await this._checkMonitors()
-      } catch (error) {
-        console.error('[WARN] scheduled function:', (error && error.stack) || error)
-      }
+    closedAlarms.forEach(alarm => {
+      byRules[alarm.rule.id] = byRules[alarm.rule.id] ?? { alarms: [] }
+      byRules[alarm.rule.id].alarms.push({
+        ...alarm,
+        url: this._generateUrl(alarm.rule.objectType, alarm.target),
+        notificationType: 'closed',
+      })
     })
+    const subject = newAlarms.size > 0 ? `Performance Alerts : new Alerts` : `Performance Alerts : end of all Alerts`
+    const { html } = await templates.mjml.transform(templates.mjml.$newAlarms({ byRules }))
+    const text = await templates.markdown.$newAlarms({ byRules })
+    return this._sendAlertEmail(subject, html, text)
   }
 
-  async configure(configuration) {
-    this._configuration = configuration
-    clearCurrentAlarms()
+  async load() {
+    if (this.#strategy) {
+      return
+    }
+    this.#monitorRuleSet = new MonitorRuleSet(this.#configuration)
+    this.#strategy = new HybridStrategy(this._xo, this.#monitorRuleSet)
+
+    this.#strategy.watch(changes => this.sendAlarmChange(changes), 60 * 1000).catch(console.error)
```

Comment:
Not sure to understand.
`HybridStrategy.watch` only take one parameter, but you are giving him 2.
Probably a misstake as `RrdHostVm.watch` take 2 params